stdlib/asm/collections/mmr.masm

```diff
@@ -0,0 +1,81 @@
+#! Computes the `ilog2(number)` and `2^(ilog2(number))`.
+#!
+#! number must be non-zero, otherwise this will error
+#!
+#! Stack transition:
+#! [number, ...] -> [ilog2, power_of_two, ...]
+export.ilog2_checked
+  push.2147483648 # power_of_two from high to low bit
+  push.0          # bit_pos from the most-signficant, `31-bit_pos` equals to ilog2
+  # stack: [bit_pos, power_of_two, number, ...]
+
+  # find the first most-significant true bit
+  dup.1 dup.3 u32checked_and eq.0 while.true
+    add.1 swap div.2 swap
+    dup.1 dup.3 u32checked_and eq.0
+  end
+
+  # compute ilog2
+  push.31 swap sub
+  # stack: [ilog2, power_of_two, number, ...]
+
+  # drop number
+  movup.2 drop
+  # stack: [ilog2, power_of_two, ...]
+end
+
+#! Loads the leaf at the absolute `pos` in the MMR.
+#!
+#! Stack transition:
+#! [pos, mmr_ptr, ...] -> [n3, n2, n1, n0, ...]
+export.get
+  # load the num_leaves of the MMR
+  dup.1 mem_load
+  # stack: [num_leaves, pos, mmr_ptr, ...]
+
+  # compute `num_leaves & pos`, this contains all peaks before `pos` (and maybe some after the owning peak)
+  dup.1 dup.1 u32checked_and
+  # stack: [before_candidates, num_leaves, pos, mmr_ptr, ...]
```

Comment:
> Is there an implicit assumption here that pos and num_leaves must be u32 values?

Not really, this could also be done with `u64` procedures, it only needs `checked_and` and `popcnt`, the former is already implemented the later is super easy to add.

I used u32 just because it was more efficient.

Added a comment to the procedure stating only u32 positions are supported.